app/controllers/application_controller.rb

```diff
@@ -1,4 +1,6 @@
 class ApplicationController < ActionController::Base
+  include SetCurrentRequestDetails
```

Comment:
Given that this is only currently used once here in the base controller I don't think it needs to be extracted to a concern. It would be less indirection if the `before_action` was inlined here. It could be written as a named method so that it still has a name, e.g. `before_action :set_current_request_details`